runtime/include/wasm_table.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/* memory also provides the table access functions */
+#define INDIRECT_TABLE_SIZE (1 << 10)
+
+struct wasm_table_entry {
+	uint32_t type_id;
+	void *   func_pointer;
+};
+
+struct wasm_table {
+	uint32_t                length;
+	uint32_t                capacity;
+	struct wasm_table_entry data[];
+};
+
+static inline struct wasm_table *
+wasm_table_allocate(size_t capacity)
+{
+	struct wasm_table *self = (struct wasm_table *)malloc(sizeof(struct wasm_table)
+	                                                      + capacity * sizeof(struct wasm_table_entry));
+
+	self->capacity = capacity;
+	self->length   = 0;
+
+	return self;
+}
+
+static inline void
+wasm_table_free(struct wasm_table *self)
+{
+	assert(self != NULL);
+	free(self);
+}
+
+static inline void *
+wasm_table_get(struct wasm_table *self, uint32_t idx, uint32_t type_id)
+{
+	assert(self != NULL);
+	assert(idx < self->capacity);
+
+	struct wasm_table_entry f = self->data[idx];
+	// FIXME: Commented out function type check because of gocr
```

Comment:
Yikes! Maybe keep it by default and have a #define or something?

Reply:
This ugly baby has been around for a while. I'll flag an issue for this and revisit this after we merge.

Reply:
Opened #289